docker/build_support/find-updated-sources

```diff
@@ -0,0 +1,44 @@
+#! /usr/bin/env bash
+set -e
+
+prog=$0
+error() {
+   echo "Whoops!  Looks like $0:$2 failed."
+   echo "Please try rerunning $prog again."
+   exit 1
+}
+trap 'error "${BASH_SOURCE}" "${LINENO}"' ERR
+
```

Comment:
Since we're using make to determine when build and test processes need to run, and make abstracts everything using files, we need to do a little bit of magic to "help" make only see files that have been recently updated.

In development, we simply find all source files (things that actually effect image build) and find the most recently updated file and present that.

In ci, we go one step further: we only find source files that were actually -changed- since the last github action merge.  This greatly reduces work done in CI.